base/special/exp.jl

```diff
@@ -114,13 +118,13 @@ function exp{T<:Union{Float32,Float64}}(x::T)
         # scale back
         if k > -significand_bits(T)
             # multiply by 2.0 first to prevent overflow, which helps extends the range
-            k == exponent_max(T) && return y*T(2.0)*T(2.0)^(exponent_max(T) - 1)
+            k == exponent_max(T) && return y * T(2.0) * T(2.0)^(exponent_max(T) - 1)
             twopk = reinterpret(T, rem(exponent_bias(T) + k, fpinttype(T)) << significand_bits(T))
             return y*twopk
         else
             # add significand_bits(T) + 1 to lift the range outside the subnormals
             twopk = reinterpret(T, rem(exponent_bias(T) + significand_bits(T) + 1 + k, fpinttype(T)) << significand_bits(T))
-            return y*twopk*T(2.0)^(-significand_bits(T) - 1)
+            return y * twopk * T(2.0)^(-significand_bits(T) - 1)
```

Comment:
I think it's better as is, since the intent of the code is a lot more clear with it being explicit same with `ldexp`  (which is why I didn't use `eps`, as it makes the scaling and explicit) 

but if they must change it would also make sense to change both 
`T(2.0)^(exponent_max(T) - 1)`
and
`T(2.0)^(-significand_bits(T) - 1)`
should either both us the `^ = Base.FastMath.pow_fast` or be evaluated ahead as constants. 

I have to admit it's a little disappointing that we have to now work around these  cases as such